### PR TITLE
Database scripts

### PR DESCRIPTION
**Pull request by jgarst**
- peewee bumped to latest version. It makes sense to fix the breaking changes before adding more database code
- fixed a bug where makedb.py would always use the default xml folder
- I can never get relative imports to work correctly, and they are discouraged in the official style guide, so I changed them to absolute.

peewee probably doesn't require the absolute latest version, but there is no quick way to check what it supports.

### CODE REVIEW

**Comment by mpacer** [comment]
@sbassi You know the most about the database code, do you have any thoughts?

**Comment by mpacer** [comment]
Where are relative imports discouraged in the official style guide? Could you provide a link?

File: allofplos/dbtoorm.py [hunk 1]
```diff
@@ -43,11 +43,11 @@ class Meta:
 
 class Plosarticle(BaseModel):
     doi = TextField(db_column='DOI', unique=True)
-    jats_type = ForeignKeyField(db_column='JATS_type_id', rel_model=Jatstype, to_field='id')
+    jats_type = ForeignKeyField(column_name='JATS_type_id', model=Jatstype, field='id')
```
Comment (sbassi) [inline allofplos/dbtoorm.py]: this change is something related to Peewee 3.5?

**Comment by mpacer** [comment]
If you mean from the PEP 8 style guide, it doesn't discourage them:

> Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path):
> 
> ```
> import mypkg.sibling
> from mypkg import sibling
> from mypkg.sibling import example
> ```
>
> However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:
>
> ```
> from . import sibling
> from .sibling import example
> ```
> 
> Standard library code should avoid complex package layouts and always use absolute imports.
> 
> Implicit relative imports should never be used and have been removed in Python 3.

File: allofplos/makedb.py [hunk 2]
```diff
@@ -122,9 +122,7 @@ class CoAuthorPLOSArticle(BaseModel):
 allfiles = Corpus(corpus_dir).files
 files = random.sample(allfiles, args.random) if args.random else allfiles
 
-for file_ in tqdm(files):
-    doi = filename_to_doi(file_)
-    article = Article(doi)
+for article in tqdm(Corpus(corpus_dir)):
```
Comment (mpacer) [inline allofplos/makedb.py]: w00t!

Comment (sbassi) [inline allofplos/makedb.py]: if you don't iterate over "files" you are not iterating over the random sample if that is the option in the command line

Comment (jgarst) [reply allofplos/makedb.py]: Thanks for catching that. The Corpus iterator is `random_article_generator`, so I'll solve this by limiting the number of articles we pull from Corpus.

One reason I neglected tests with this pull request is that I'm uncertain what you want to enshrine as permanent. Did you imagine the database code combining with the xml based classes, or staying as a separate utility?

Also, sorry for the lack of tests.

**Comment by jgarst** [comment]
I guess I interpret the "Absolute imports are recommended" part more strongly than you do. Not that it matters, as I also don't think that pep8 should be the last word on style.

The actual problem that part solved was a `No module named '__main__.corpus'` error when running the script from the command line. I think there is another way to solve that issue by getting into some __init__.py weeds? I'm happy to do that, but will wait until a member of the core team explicitly tells me to, as M Pacer also indicated she was enthusiastic enough to merge that part as is.

File: allofplos/makedb.py (hunk 2 above)
Comment (sbassi) [reply allofplos/makedb.py]: I never thought of this utility as something that can be integrated with xml classes, so as a separate script is OK for me.

Comment (sbassi) [reply allofplos/makedb.py]: waiting for the commit with "I'll solve this by limiting the number of articles we pull from Corpus."

**Comment by jgarst** [comment]
Sorry to make you wait for a simple change.

I spent a few minutes trying to figure out how to test this nicely, but it is tricky while these scripts don't have functions that I can pass examples into.

I don't want to refactor out from under you without chatting first, either over chat or in person. Otherwise I'll refocus on command line issues and examples of using this package.

**Comment by sbassi** [comment]
@jgarst I can't find the change related to "fixed a bug where makedb.py would always use the default xml folder"

**Comment by jgarst** [comment]
This is the line with the bug on master, https://github.com/PLOS/allofplos/blob/master/allofplos/makedb.py#L127

It creates the article without using the directory the user specifies, so it will always look in allofplos_xml, instead of using either allofplos_xml or the starter directory. This pull request dodges that by using the article iterator from Corpus directly.

**Comment by sbassi** [comment]
I see it now, thank you.

**Pull request merged**